PhysicsTools/NanoAOD/python/taus_cff.py

```diff
@@ -158,25 +161,29 @@ def _tauId8WPMask(pattern,doc):
 
 tauTable.variables = _variablesMiniV2
 
+(run2_nanoAOD_92X | run2_nanoAOD_94XMiniAODv2 | run2_nanoAOD_94X2016 | run2_nanoAOD_102Xv1 | run2_nanoAOD_106Xv1).toModify(tauTable,
+                 variables = cms.PSet(tauTable.variables, _mvaAntiEVars, _mvaIsoVars2015Reduced, _mvaIsoVars2017v1, _mvaIsoVars2017v2)
+    )
 for era in [run2_nanoAOD_94XMiniAODv1,]:
     era.toModify(tauTable,
                  variables = _variablesMiniV1
     )
 run2_miniAOD_80XLegacy.toModify(tauTable,
                                 variables = _variables80X
 )
+for era in [run2_miniAOD_80XLegacy, run2_nanoAOD_92X, run2_nanoAOD_94XMiniAODv1, run2_nanoAOD_94XMiniAODv2, \
+            run2_nanoAOD_94X2016, run2_nanoAOD_102Xv1, run2_nanoAOD_106Xv1]:
+    era.toModify(tauTable.variables,
+                 idDecayMode = Var("tauID('decayModeFinding')", bool),
+                 idDecayModeNewDMs = Var("tauID('decayModeFindingNewDMs')", bool),
+                 idDecayModeOldDMs = None
+    )
 run2_miniAOD_80XLegacy.toModify(tauTable.variables,
                                 idAntiEleDeadECal = None
 )
 
 (run2_nanoAOD_106Xv2 | run2_miniAOD_devel).toModify(tauTable.variables,
-                            idAntiEleDeadECal = Var("tauID('againstElectronDeadECAL')", bool, doc = "Anti-electron dead-ECal discriminator"),
-                            rawAntiEle2018 = Var("tauID('againstElectronMVA6Raw')", float, doc= "Anti-electron MVA discriminator V6 raw output discriminator (2018)", precision=10),
-                            rawAntiEleCat2018 = Var("tauID('againstElectronMVA6category')", int, doc="Anti-electron MVA discriminator V6 category (2018)"),
-                            idAntiEle2018 = _tauId5WPMask("againstElectron%sMVA6", doc= "Anti-electron MVA discriminator V6 (2018)"),
-                            rawAntiEle = Var("tauID('againstElectronMVA6Raw2015')", float, doc= "Anti-electron MVA discriminator V6 raw output discriminator (2015)", precision=10),
-                            rawAntiEleCat = Var("tauID('againstElectronMVA6category2015')", int, doc="Anti-electron MVA discriminator V6 category (2015"),
-                            idAntiEle = _tauId5WPMask("againstElectron%sMVA62015", doc= "Anti-electron MVA discriminator V6 (2015)")
+                            idAntiEleDeadECal = Var("tauID('againstElectronDeadECAL')", bool, doc = "Anti-electron dead-ECal discriminator")
```

Comment:
this should be done like in master:

one set the future as default, and then rewire the past.

https://github.com/cms-tau-pog/cmssw/blob/9aef44e5dbcc067ab65d21177b3ef40426426d52/PhysicsTools/NanoAOD/python/taus_cff.py#L81

https://github.com/cms-tau-pog/cmssw/blob/9aef44e5dbcc067ab65d21177b3ef40426426d52/PhysicsTools/NanoAOD/python/taus_cff.py#L176-L179

Reply:
alright, to be updated. As I said above, I tried to keep changes wrt. present implementation small. But in fact I was wondering why this was not consistent with the convention you are describing.

Reply:
I will adapt the logic for idAntiEleDeadECal as well (while the quantity itself was not part of the clean up so far)